programs/lz4io.c

```diff
@@ -395,8 +404,50 @@ typedef struct {
     void*  dstBuffer;
     size_t dstBufferSize;
     LZ4F_compressionContext_t ctx;
+    LZ4F_CDict* cdict;
 } cRess_t;
 
+static void LZ4IO_loadDict(char **dictionaryBuffer, size_t *dictionarySize) {
+    FILE* dictionaryFile;
+    size_t blockSize = 64 KB;
+    size_t dictionaryBufferSize = blockSize;
+    size_t readSize;
+    *dictionarySize = 0;
+    *dictionaryBuffer = malloc(dictionaryBufferSize);
+
+    if (!g_dictionaryFilename) {
+        EXM_THROW(25, "Dictionary error : no filename provided");
+    }
+
+    dictionaryFile = LZ4IO_openSrcFile(g_dictionaryFilename);
+
+    do {
+        if (*dictionarySize + blockSize > dictionaryBufferSize) {
+            dictionaryBufferSize *= 2;
```

Comment:
An LZ4 dictionary cannot be larger than 64KB
(or more precisely, in this case, only the last 64KB are loaded).
So it's not useful to load more than 64 KB.

If the dictionary file is larger than 64 KB (which can easily be the case, since zstd dictionary generator creates 110 KB files by default), load only the last 64 KB into memory buffer.

It would also be useful to set some kind of "limit" : 
if someone provides a 2 GB file as dictionary, it's likely an error.

Reply:
Yeah, I thought about doing that. So we don't necessarily know the size of the file we're reading. (It could be a pipe.) So we probably have to read the whole thing anyways.

But I guess I could read into a circular buffer instead.